boa_engine/src/syntax/parser/cursor/mod.rs

```diff
@@ -83,6 +83,30 @@ where
         self.buffered_lexer.peek(skip_n, true, interner)
     }
 
+    #[inline]
+    pub(super) fn peek_or_abrupt(
+        &mut self,
+        skip_n: usize,
+        interner: &mut Interner,
+    ) -> Result<&Token, ParseError> {
+        self.buffered_lexer
+            .peek(skip_n, true, interner)?
+            .ok_or(ParseError::AbruptEnd)
+    }
+
+    #[inline]
+    pub(super) fn peek_kind(
+        &mut self,
+        skip_n: usize,
+        interner: &mut Interner,
+    ) -> Result<&TokenKind, ParseError> {
+        Ok(self
+            .buffered_lexer
+            .peek(skip_n, true, interner)?
+            .ok_or(ParseError::AbruptEnd)?
+            .kind())
+    }
+
```

Comment:
I see that the `peek_or_abrupt()` function is only being used once, and the `peek_kind()` function only twice. Do we really need them? It might be easy to use the existing functions. What do you think?

Reply:
I just found these two patterns are frequently used in other places as well, so I made these functions for other uses. I just didn't want to update existing code in this PR. So, yeah it's up to you. I don't mind removing them and using existing functions.

Reply:
I would re-use the existing ones. The inclusion of new functions (and using them throughout the code) should come in a different PR.